apiserver/facades/client/highavailability/highavailability.go

```diff
@@ -34,13 +34,19 @@ type NodeService interface {
 	CurateNodes(context.Context, []string, []string) error
 }
 
+// MachineSaver instances save a machine to dqlite state.
+type MachineSaver interface {
+	Save(context.Context, string) error
```

Comment:
Given that services are about domains rather than entities, method names should be explicit, indicating that you are specifically adding or updating a machine.